### PR TITLE
Added Cabal flags for benchmark suite

### PR DESCRIPTION
**Pull request by mmhat**
There is one flag for each effect library. Those flag control whether
that benchmark is included or not.

### CODE REVIEW

**Comment by arybczak** [comment]
Why do you want such granularity? Wouldn't one flag for all 3 libraries be enough?

**Comment by mmhat** [comment]
I used those flags to compare effectful to X one by one. But I guess one flag is sufficient.
BTW: Should we run the other benchmarks by default or not?

**Comment by arybczak** [comment]
> BTW: Should we run the other benchmarks by default or not?

Yes.

Please don't gate `mtl` behind a flag, it's unnecessary as it's a GHC boot dependency.

Actually, what is the advantage of having this flag? It's not like the other effect libraries pull in tons of dependencies, they're pretty lightweight.

**Comment by mmhat** [comment]
The primary motivation behind this flag is the following: When working on effectful I'd like to spot performance regression. The other benchmarks are not interesting (unless one is working on the benchmark suite itself); They just add some noise to the benchmark outputs and lead to longer development cycles. I don't care so much about the dependencies.

That's why I gated `mtl` as well.

File: bench/FileSizes.hs [hunk 1]
```diff
@@ -6,7 +6,9 @@ module FileSizes where
 import Control.Exception
 import Control.Monad.IO.Class
 import Data.IORef
+#ifdef VERSION_polysemy
 import Data.Tuple
```
Comment (arybczak) [inline bench/FileSizes.hs:10]: AFAIK code for `eff` also uses this. You can remove this and add `eff_swap` and `poly_swap` functions below.

File: bench/Countdown.hs [hunk 2]
```diff
@@ -2,7 +2,9 @@
 module Countdown where
 
 import Control.Monad.ST
+#ifdef VERSION_mtl
 import Data.Functor.Identity
```
Comment (arybczak) [inline bench/Countdown.hs:6]: Please move that to the block for `mtl` imports below.

**Comment by arybczak** [comment]
> When working on effectful I'd like to spot performance regression

Fair enough :+1:

File: bench/FileSizes.hs (hunk 1 above)
Comment (mmhat) [reply bench/FileSizes.hs]: Done.

File: bench/Countdown.hs (hunk 2 above)
Comment (mmhat) [reply bench/Countdown.hs]: Oh, I missed that one! Done.

**Pull request merged**